handlers/photo-audio-haptics-handler/src/server.ts

```diff
@@ -0,0 +1,267 @@
+import handlerResponseJSON from "./schemas/handler-response.schema.json";
+import definitionsJSON from "./schemas/definitions.json";
+import ttsRequestJSON from "./schemas/services/tts/segment.request.json";
+import ttsResponseJSON from "./schemas/services/tts/segment.response.json";
+import descriptionJSON from "./schemas/services/supercollider/tts-description.schema.json";
+import segmentJSON from "./schemas/services/supercollider/tts-segment.schema.json";
+import rendererDefJSON from "./schemas/renderers/definitions.json";
+import textJSON from "./schemas/renderers/text.schema.json";
+import photoAudioHapticsJSON from "./schemas/renderers/photoaudiohaptics.schema.json";//"./photoaudiohaptics.schema.json";
+
+import * as utils from "./utils";
+
+const ajv = new Ajv({
+    "schemas": [querySchemaJSON,
+        handlerResponseJSON,
+        definitionsJSON,
+        ttsRequestJSON,
+        ttsResponseJSON,
+        descriptionJSON,
+        segmentJSON,
+        rendererDefJSON,
+        photoAudioHapticsJSON]
+});
+
+const app = express();
+const port = 80;
+const scPort = 57120;
+const filePrefix = "/tmp/sc-store/photo-audio-haptics-handler-";
+
+app.use(express.json({ limit: process.env.MAX_BODY }));
+
+app.post("/handler", async (req, res) => {
+    // Validate the request data (just in case)
+    if (!ajv.validate("https://image.a11y.mcgill.ca/request.schema.json", req.body)) {
+        console.warn("Request did not pass the schema!");
+        res.status(400).json(ajv.errors);
+        return;
+    }
+
+    const renderings: any = [];
+
+    // *******************************************************
+    // Check for preprocessor data
+    // *******************************************************
+    const preprocessors = req.body["preprocessors"];
+    const preSecondCat = preprocessors["ca.mcgill.a11y.image.preprocessor.secondCategoriser"];
+    const preSemSeg = preprocessors["ca.mcgill.a11y.image.preprocessor.semanticSegmentation"]
+    const preObjDet = preprocessors["ca.mcgill.a11y.image.preprocessor.objectDetection"]
+    const preGroupData = preprocessors["ca.mcgill.a11y.image.preprocessor.grouping"];
+
+    if (!preObjDet && !preSemSeg && !preGroupData) {
+        console.debug("No preprocessor data available: can't render!");
+        const response = utils.generateEmptyResponse(req.body["request_uuid"]);
+
+        if (ajv.validate("https://image.a11y.mcgill.ca/handler-response.schema.json", response)) {
+            res.json(response);
+        } else {
+            console.error("Failed to generate a valid empty response!");
+            console.error(ajv.errors);
+            res.status(500).json(ajv.errors);
+        }
+        return;
+    }
+
+    // *******************************************************
+    // Check for renderer availability
+    // *******************************************************
+    const hasText = req.body["renderers"].includes("ca.mcgill.a11y.image.renderer.Text");

```

Comment:
This is only used for the check on line 93 and no actual rendering is reported. Best to remove this line and any reference to the Text renderer since it is unused.